lua/toggleterm.lua

```diff
@@ -21,6 +21,10 @@ local AUGROUP = "ToggleTermCommands"
 -----------------------------------------------------------
 local M = {}
 
+---@type number[]
+-- A list of terminal IDs that are saved from the view on smart toggle.
+local terminal_view = {}
```

Comment:
Seems at this point that there should maybe be a table called `state` which holds the various bits of internal state in this file, so it's easier to see at a glance what state is being depended on centralise changing it

Reply:
Actually just noticed this is the toggleterm file, I wonder if the state can be held in the terminal file which is where I think of as where the internal terminal logic is done, or perhaps in the ui file which is where I think of the ui helpers and perhaps state also living in any case the main issue is having state in different places rather than centralised

Reply:
good point, I was thinking about this, and since it's directly tied to the `toggle` functionality I kept it in the `toggleterm` module. But I get what you mean, and I'll transfer saving/opening logic to `terminal` module and we can keep the state there as well.

Reply:
done, I've moved this logic/state to `ui` module